docs/how-to/integrate-external-libraries.md

```diff
@@ -0,0 +1,252 @@
+
+- **Source or Pre-compiled**: Do you have the source code for the library, or is it pre-compiled (installed on your system, or `.a / .so` files available)?
+- **Build System** _(only applies if you have the source)_: Does the library use CMake, or another build system?
+- **CMake integration**: Does the library provide CMake configuration files such as `<libName>Config.cmake`? (refer to the [CMake documentation](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html) for more information)
+
+The following summarizes recommendation on which approach to use based on these considerations:
+
+1. **Pre-compiled**:
+  a) If you only have a pre-compiled library and no CMake integration, use `MOD_DEPS` (Approach 1).
+  b) If you have a pre-compiled library along with CMake integration, use `find_package()` (Approach 3).
+2. **Source**:
+  a) If the library uses CMake, use `FetchContent / add_subdirectory` (Approach 2).
+  b) If the library does not use CMake but provides CMake integration, use `find_package()` (Approach 3) or `ExternalProject_Add()` (Approach 4).
+  c) If the library does not use CMake and provides no CMake integration, use `ExternalProject_Add()` (Approach 4).
+
+Note that these are mere recommendations, and that you may choose to use a different approach based on your project's needs or the library's requirements. The following sections will cover each of these approaches in detail, along with their benefits and drawbacks.
+
+---
+
+## Approach 1: Pre-compiled library files
+
+|     |     |
+| --- | --- |
+| **Benefits** | Easy to set up; no additional build time |
+| **Drawbacks** | Not portable across platforms |
+| **Considerations** | May be provided by a vendor or third-party |
+
+A pre-compiled library is a library that has already been compiled and is ready to be used, often named `lib<libName>.a` or `lib<libName>.so`. There are many ways to obtain pre-compiled libraries, such as downloading them from a vendor repository or building them from source yourself.
+
+To integrate a pre-compiled library, you need to add the path of that library file to `MOD_DEPS` of the module(s) that depend on it (MOD_DEPS: module dependencies). The following example demonstrates how to integrate the OpenSSL `libcrypto` library into an F´ wrapper component:
+
+```cmake
+set(SOURCE_FILES
+    "${CMAKE_CURRENT_LIST_DIR}/OpenSslWrapper.fpp"
+    "${CMAKE_CURRENT_LIST_DIR}/OpenSslWrapper.cpp"
+)
+set(MOD_DEPS
+    ${FPRIME_PROJECT_ROOT}/lib/openssl/libcrypto.a
+)
+register_fprime_module()
+target_include_directories(${FPRIME_CURRENT_MODULE} PUBLIC "${FPRIME_PROJECT_ROOT}/lib/openssl/include")
+```
+
+This assumes that the `libcrypto.a` is available in the `lib/openssl/` directory of your F´ project root, and that the necessary header files are in `lib/openssl/include/`.
+
+---
+
+## Approach 2: CMake library with FetchContent or add_subdirectory
```

Comment:
Split this into two approaches I think.  As add_subdirectory is exactly for CMake libraries added to your repos.  FetchContent is for else.

Reply:
You mention it below, but it is not highlighted.